### PR TITLE
[docs, channel-client] Add autogenerated API docs to our website

### PR DESCRIPTION
**Pull request by geoknee**
I spent quite a bit of time trying TypeDoc + `typedoc-plugin-markdown` with docusarus theme. The output was not great, so I spent some more time trying to write my own theme. I gave up on this. 

I also briefly considered JSDoc. 

Then I tried api-extractor, which worked pretty well!

~This branch mostly a POC at the moment. It has thrown up a few questions about our overall approach.~

### CODE REVIEW

**Comment by geoknee** [comment]
Preview here https://5f106017597266d269ed975d--nitro-protocol.netlify.app/docs/next/channel-client-api/channel-client.channelclient

File: packages/channel-client/src/channel-client.ts [hunk 1]
```diff
@@ -94,10 +146,28 @@ export class ChannelClient implements ChannelClientInterface {
...
+   */
   async joinChannel(channelId: string): Promise<ChannelResult> {
     return this.provider.send({method: 'JoinChannel', params: {channelId}});
   }
 
+  /**
+   * Updates the state of a channel
+   *
+   * @param channelId id for the state channel
+   * @param participants Array of Participants for this channel TODO not clear why this is here, as it must be constant once the channel has its id.
```
Comment (geoknee) [inline packages/channel-client/src/channel-client.ts:165]: ❓

File: packages/channel-client/src/channel-client.ts [hunk 2]
```diff
@@ -159,10 +272,25 @@ export class ChannelClient implements ChannelClientInterface {
...
   async getBudget(hubAddress: string): Promise<DomainBudget> {
     return this.provider.send({method: 'GetBudget', params: {hubAddress}});
   }
 
+  /**
+   * Requests the funds to be withdrawn from this site's ledger channel
+   *
+   * @param hubAddress The address of a state channel hub
+   * @param hubOutcomeAddress An ethereum account that the hub's funds will be paid out to TODO this doesn't make sense
+   * @returns A promise that resolves to a DomainBudget.
```
Comment (geoknee) [inline packages/channel-client/src/channel-client.ts:291]: ❓ 
(for `closeAndWithdraw`, I don't understand why I should get to say anything about the hub's outcome address)

Comment (geoknee) [reply packages/channel-client/src/channel-client.ts:291]: This probably also does not belong, but we should double check to see what (if anything) the wallet does with it.

File: packages/channel-client/src/channel-client.ts (hunk 1 above)
Comment (geoknee) [reply packages/channel-client/src/channel-client.ts:165]: Agreed that this does not belong here.

File: packages/channel-client/src/channel-client.ts (hunk 2 above)
Comment (geoknee) [reply packages/channel-client/src/channel-client.ts:291]: TODO

File: packages/channel-client/src/channel-client.ts (hunk 1 above)
Comment (geoknee) [reply packages/channel-client/src/channel-client.ts:165]: #2296

**Comment by snario** [comment]
It would be nice if the ways an API call can fail were visible:

https://5f106017597266d269ed975d--nitro-protocol.netlify.app/docs/next/channel-client-api/channel-client.channelclient.updatechannel

File: packages/channel-client/src/channel-client.ts (hunk 2 above)
Comment (geoknee) [reply packages/channel-client/src/channel-client.ts:288]: #2306

**Comment by geoknee** [comment]
> We could instead type `ChannelResult | UpdateChannelError` and type those errors.

Agreed, but this is way out of scope for this PR, right?

**Comment by snario** [comment]
> Agreed, but this is way out of scope for this PR, right?

Yeah, but the existing docs manually add the errors. e.g., https://api.statechannels.org/#join-channel

**Comment by geoknee** [comment]
> Yeah, but the existing docs manually add the errors. e.g., https://api.statechannels.org/#join-channel

Ah, those docs are for the JSON-RPC interface, which these docs are not necessarily going to replace on their own. 

Regardless,  I think it should be possible to thread those errors through into these docs. 

In fact they are in there already, after a fashion https://5f106017597266d269ed975d--nitro-protocol.netlify.app/docs/next/channel-client-api/channel-client.userdeclinederrorcode

I'll see what I can do to make that nicer.

**Comment by geoknee** [comment]
I think we should merge this now (establishes the docgen pipeline) and I'll keep improving the annotations and integration into the website.

**Comment by geoknee** [comment]
Latest preview: https://5f1801eb24378972643f1372--nitro-protocol.netlify.app/docs/next/channel-client-api/channel-client.channelclient

**Pull request merged**